text/0750-feature-flag-advisor.md

```diff
@@ -0,0 +1,343 @@
+  Setting all feature flags to the recommended values...
+
+  Resynthesizing... 
+  
+  Here is the difference:
+  - S3 Bucket:
+  -    Properties:
+  -      `BucketEncryption``:`` ``None`
+  + S3 Bucket:
+  +    Properties:
+  +      BucketEncryption: 
+  +        `ServerSideEncryptionConfiguration``:`
+  +          ...
+  
+```
+
+#### I only want to change one feature flag to its recommended state
+
+To simply modify one feature flag to our recommended state, you can run `cdk flags --set-recommended --flagName="#FLAGNAME#"`.
+
+```
+> cdk flags --set-recommended --flagName="#FLAGNAME#"
+    Feature Flag                              Recommended Value            User Value
+  * @aws-cdk/...                              true                         false
+  Setting all feature flags to the recommended values...
+
+  Resynthesizing... 
+  
+  Here is the difference:
+  - S3 Bucket:
+  -    Properties:
+  -      `BucketEncryption``:`` ``None`
+  + S3 Bucket:
+  +    Properties:
+  +      BucketEncryption: 
+  +        `ServerSideEncryptionConfiguration``:`
+  +          ...
+  
+```
+
+#### I want to modify one feature flag to a different value
+
+To modify one feature flag, first we view its information.
+
+```
+> cdk flags "@aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021"
+    Description: Enable this feature flag to have cloudfront distributions use the security policy TLSv1.2_2021 by default.
+    Recommended Value: true
+    User Value: true
+```
+
+Additionally, to modify the state, you can run `cdk flags --set --flagName="#FLAGNAME#"`. This will return a list of values for you to choose from.
+
+```
+> cdk flags "@aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021" --set
+  What value would you like to change "@aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021" to?
+  > true
+    false
+> true
+
+Resynthesizing... 
+
+Here is the difference:
+- S3 Bucket:
+-    Properties:
+-      `BucketEncryption``:`` ``None`
++ S3 Bucket:
++    Properties:
++      BucketEncryption: 
++        `ServerSideEncryptionConfiguration``:`
++          ...
+```
+
+#### I want to revert my changes
+
+After performing any of the above operations, you will be shown the difference in CloudFormation templates and asked if you would like to
+keep the changes as shown below. If you choose to revert the changes, you are given options to revert all flags to its previous states
+or just a specific one. There will be a list of flags for you to choose from to revert.
+
+````
+Would you like to accept these changes? (Accept/revert)
+> revert
+Would you like to revert the entire feature flag configuration to its previous state or just revert the specific change(s)?
+[1] Revert all changes -> Restore all flags to their previous state
+[2] Revert only changes to one feature flag -> Undo a specific flag change
+[3] Exit without making any changes -> Keep your current configuration as is
```

Comment:
What is the state of my current configuration? is it after applying the changes? i.e as I was happy with the result?